orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/persistence/jedis/RedisExecutionRepository.java

```diff
@@ -708,11 +779,39 @@ private void deleteInternal(JedisCommands c, ExecutionType type, String id) {
       .subscribeOn(scheduler);
   }
 
-  protected static String alljobsKey(ExecutionType type) {
+  protected ImmutableMap<String, RedisClientDelegate> fetchKey(String id) {
```

Comment:
I don't typically recommend using Apache Commons, but it's already on the classpath, so I'd definitely refactor to use `ImmutablePair` instead. That will simplify the usage of this method quite dramatically:

```java
import org.apache.commons.lang3.tuple.ImmutablePair;

ImmutablePair pair = ImmutablePair.of(pipelineKey(redisClientDelegate.id), redisClientDelegate);
// pair.getLeft(), pair.getRight()
ImmutablePair pair = ImmutablePair.nullPair();
```

Reply:
Creating a stream just to fish out the key / value from a map would be quite a lot of overhead.

Reply:
👍